plugins/woocommerce/src/Blocks/BlockTypes/ProductImage.php

```diff
@@ -235,20 +228,31 @@ protected function render( $attributes, $content, $block ) {
 			)
 		);
 
-		if ( $product ) {
-			return sprintf(
-				'<div %1$s>
-					%2$s
-				</div>',
-				$wrapper_attributes,
-				$this->render_anchor(
-					$product,
-					$this->render_on_sale_badge( $product, $parsed_attributes ),
-					$this->render_image( $product, $parsed_attributes ),
-					$parsed_attributes
-				)
-			);
+		if ( ! $product instanceof \WC_Product ) {
+			return;
+		}
 
+		// Support older versions of the block that use `showSaleBadge` instead of inner blocks.
+		$sale_badge = '';
+		if ( isset( $parsed_attributes['showSaleBadge'] ) ) {
+			$sale_badge = $this->render_on_sale_badge( $product, $parsed_attributes );
 		}
+
+		return sprintf(
+			'<div %1$s>
+				%2$s
+				<div class="wc-block-components-product-image-inner-blocks">
+					%3$s
+				</div>
+			</div>',
+			$wrapper_attributes,
+			$this->render_anchor(
+				$product,
+				$sale_badge,
+				$this->render_image( $product, $parsed_attributes ),
```

Comment:
I think we follow the cover block here to manually style the image, we can set the `object-fit` property to `cover` and control the dimension using the wrapper element. We can also leverage the Supports API here to eliminate some of the manual style processing.

Reply:
Talking about block support, we need move block supports declaration to `block.json` and remove `get_block_type_supports()` (which does nothing since the introduction of Product Image `block.json` file).